addons/base_vat/models/res_partner.py

```diff
@@ -151,6 +156,8 @@ def check_vat(self):
             failed_check = False
             vat_country_code, vat_number = self._split_vat(partner.vat)
             vat_guessed_country = self.env['res.country'].search([('code', '=', vat_country_code.upper())])
```

Comment:
Keeping the same name for this variable makes its use confusing.
Rename it something like vat_has_legit_country_code